Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp

```diff
@@ -750,12 +750,82 @@ class OMGIRGenerator {
         return WTFMove(m_exceptionHandlers);
     }
 
+    struct AbstractHeap {
+        enum Kind : uint8_t {
+            Globals,
+            WasmMemory,
+            CallSiteIndex,
+            WasmTables,
+            WasmImports,
+            InstanceFields,
+            TableFields,
+            FuncRefTableFields,
+            FunctionFields,
+            NumKinds
+        };
+
+        Kind kind;
+        unsigned offset : 24;
```

Comment:
Does this actually make a bitfield? Technically, I thought all members have to be the same type. Although maybe that's changed in newer C++ versions...

Reply:
I was wondering about something similar and plugged it into compiler explorer, the sizeof the whole structure does come out to 4B so it looks to be working as expected.

Reply:
Good catch...that's weird that it came out as 4B, but I'll define `Kind : unsigned` to guarantee it.